### PR TITLE
Create codeql-analysis.yml

### PR DESCRIPTION
**Pull request by jirikuncar**

### CODE REVIEW

**Comment by ganeshkumarsv** [comment]
@jirikuncar Gentle ping to merge the PR. It is a part of Datadog compliance requirements to regularly perform code-scanning on all the customer installed code.

File: .github/workflows/codeql-analysis.yml [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+        # CodeQL supports [ 'cpp', 'csharp', 'go', 'java', 'javascript', 'python', 'ruby' ]
+        # Learn more about CodeQL language support at https://git.io/codeql-language-support
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v2
+
+    # Initializes the CodeQL tools for scanning.
+    - name: Initialize CodeQL
+      uses: github/codeql-action/init@v1
```
Comment (therve) [inline .github/workflows/codeql-analysis.yml:45]:
```suggestion
      uses: github/codeql-action/init@v2
```

File: .github/workflows/codeql-analysis.yml [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+        languages: ${{ matrix.language }}
+        # If you wish to specify custom queries, you can do so here or in a config file.
+        # By default, queries listed here will override any specified in a config file.
+        # Prefix the list here with "+" to use these queries and those in the config file.
+        # queries: ./path/to/local/query, your-org/your-repo/queries@main
+
+    # Autobuild attempts to build any compiled languages  (C/C++, C#, or Java).
+    # If this step fails, then you should remove it and run the build manually (see below)
+    - name: Autobuild
+      uses: github/codeql-action/autobuild@v1
```
Comment (therve) [inline .github/workflows/codeql-analysis.yml:56]:
```suggestion
      uses: github/codeql-action/autobuild@v2
```

File: .github/workflows/codeql-analysis.yml [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+    # ✏️ If the Autobuild fails above, remove it and uncomment the following three lines
+    #    and modify them (or add more) to build your code if your project
+    #    uses a compiled language
+
+    #- run: |
+    #   make bootstrap
+    #   make release
+
+    - name: Perform CodeQL Analysis
+      uses: github/codeql-action/analyze@v1
```
Comment (therve) [inline .github/workflows/codeql-analysis.yml:70]:
```suggestion
      uses: github/codeql-action/analyze@v2
```

**Pull request merged**